lib/client.js

```diff
@@ -0,0 +1,635 @@
+                if (!dcObj[v.datacenter]) {
+                    dcObj[v.datacenter] = [];
+                }
+
+                dcObj[v.datacenter].push(v);
+            }
+        }
+
+        res.forEach(sortByDatacenter.bind(self, self.utilization));
+
+        /*
+         * We just defer to the next tick so we're not tying
+         * up the event loop to sort a lot if the list is large
+         */
+        setImmediate(sortAndStoreDcs.bind(self, dcObj, opDcObj));
+    });
+}
+
+/*
+ * url: storinfo service URL
+ * cueballOpts: object containing required cueball options
+ */
+function StorinfoClient(opts) {
+    assert.object(opts, 'opts');
+    assert.optionalObject(opts.log, 'log');
+    assert.optionalNumber(opts.pollInterval, 'pollInterval');
+    assert.optionalBool(opts.multiDC, 'multiDC');
+    assert.optionalNumber(opts.defaultMaxStreamingSizeMB,
+        'defaultMaxStreamingSizeMB');
+    assert.optionalNumber(opts.maxUtilizationPct, 'maxUtilizationPct');
+    assert.bool(opts.standalone, 'standalone');
+
+    this.log = opts.log || bunyan.createLogger({ name: 'storinfo' });
+
+    /*
+     * If we're not running in standalone mode, create the restify client.
+     */
+    if (!opts.standalone) {
+        assert.string(opts.url, 'url');
+        assert.object(opts.cueballOpts, 'cueballOpts');
+
+        opts.cueballOpts.log = this.log;
+        var clientOpts = {
+            url: opts.url,
+            agent: new cueball.HttpAgent(opts.cueballOpts),
+            log: this.log
+        };
+
+        this.http = restify.createJsonClient(clientOpts);
+    }
+    this.standalone = opts.standalone;
+
+    EventEmitter.call(this);
+
+    /*
+     * The dcSharkMap is an object that maps datacenter names to an array of
+     * sharks sorted by available storage capacity that are all at or below the
+     * storage utilization threshold for normal manta requests.
+     */
+    this.dcSharkMap = null;
+    /*
+     * The operatorDcSharkMap is an object that maps datacenter names to an
+     * array of sharks sorted by available storage capacity that are all at or
+     * below the storage utilization threshold for operator manta requests.
+     */
+    this.operatorDcSharkMap = null;
+
+    this.dcIndex = -1;
+    this.multiDC = opts.multiDC === undefined ? true : opts.multiDC;
+    this.datacenters = null;
+
+    if (opts.pollInterval !== undefined && !opts.standalone) {
+        this.pollInterval = (opts.pollInterval * 1000);
+        this.defMaxSizeMB = opts.defaultMaxStreamingSizeMB ||
+            DEF_MAX_STREAMING_SIZE_MB;
+        this.utilization = opts.maxUtilizationPct ||
+            DEF_MAX_PERCENT_UTIL;
+        setImmediate(doPoll.bind(this));
+    }
+}
+util.inherits(StorinfoClient, EventEmitter);
+
+StorinfoClient.prototype.close = function close() {
+    if (this.pollTimer) {
+        clearTimeout(this._storageTimer);
+    }
+    if (!this.standalone) {
+        this.http.close();
```

Comment:
Why not check `this.http` for that guard?

Reply:
Fixed.